sources/packages/backend/libs/integrations/src/sfas-integration/sims-to-sfas.processing.service.ts

```diff
@@ -41,25 +41,53 @@ export class SIMSToSFASProcessingService {
     const simsToSFASStudents = new SIMSToSFASStudents();
     processSummary.info("Get all the students with updates.");
     // Get all the students with updates in student related data.
-    // TODO: SIMS to SFAS - Get all the students with updates in application and restriction related data.
     const studentIds = await this.simsToSFASService.getAllStudentsWithUpdates(
       modifiedSince,
       modifiedUntil,
     );
 
-    // Append the students with student and student related data updates.
-    // TODO: SIMS to SFAS - Append the student ids of students with updates in application and restriction related data.
-    // When application and restriction updates are retrieved, the respective
-    // student ids of applications and restrictions should be appended.
+    const applicationRecordsPromise =
+      this.simsToSFASService.getAllApplicationsWithUpdates(
+        modifiedSince,
+        modifiedUntil,
+      );
+
+    const restrictionRecordsPromise =
+      this.simsToSFASService.getAllRestrictionWithUpdates(
+        modifiedSince,
+        modifiedUntil,
+      );
+
+    processSummary.info(
+      "Get all the applications and restrictions with updates.",
+    );
+
+    const [applicationRecords, restrictionRecords] = await Promise.all([
+      applicationRecordsPromise,
+      restrictionRecordsPromise,
+    ]);
+
+    const applicationStudentIds = applicationRecords.map(
+      (application) => application.studentId,
+    );
+    const restrictionStudentIds = restrictionRecords.map(
+      (restriction) => restriction.student.id,
+    );
+
```

Comment:
Missed to mention in the last round of review. Please log the count of applications and restriction records found.

```ts
processSummary.info(
      `Found ${applicationRecords.length} application(s) and ${restrictionRecords.length} restriction(s) with updates.`,
    );
```